collector/generic_collector.go

```diff
@@ -57,6 +62,63 @@ func (collector *GenericCollector) Name() string {
 
 //Returns collected metrics and the next collection time of the collector
 func (collector *GenericCollector) Collect() (time.Time, []v2.Metric, error) {
-	//TO BE IMPLEMENTED
-	return time.Now(), nil, nil
+	minNextColTime := collector.configFile.MetricsConfig[0].PollingFrequency
+	for _, metricConfig := range collector.configFile.MetricsConfig {
+		if metricConfig.PollingFrequency < minNextColTime {
+			minNextColTime = metricConfig.PollingFrequency
+		}
+	}
+	currentTime := time.Now()
+	nextCollectionTime := currentTime.Add(time.Duration(minNextColTime * time.Second))
+
+	uri := collector.configFile.Endpoint
+	response, err := http.Get(uri)
+	if err != nil {
+		return nextCollectionTime, nil, err
+	}
+
+	pageContent, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nextCollectionTime, nil, err
+	}
+
+	metrics := make([]v2.Metric, len(collector.configFile.MetricsConfig))
+
+	for ind, metricConfig := range collector.configFile.MetricsConfig {
+		regex, err := regexp.Compile(metricConfig.Regex)
+		if err != nil {
+			return nextCollectionTime, nil, err
+		}
+
+		matchString := regex.FindStringSubmatch(string(pageContent[:]))
+		if matchString != nil {
+			if strings.Contains(matchString[1], ".") {
+				regVal, _ := strconv.ParseFloat(strings.Trim(matchString[1], " "), 64)
+				metricValue := make([]v2.FloatPoint, 1)
+				metricValue[0].Value = regVal
+				metricValue[0].Timestamp = currentTime
+				metrics[ind].FloatPoints = metricValue
+				metrics[ind].IntPoints = nil
+
+			} else {
+				regVal, _ := strconv.ParseInt(strings.Trim(matchString[1], " "), 10, 64)
+				metricValue := make([]v2.IntPoint, 1)
+				metricValue[0].Value = regVal
+				metricValue[0].Timestamp = currentTime
+				metrics[ind].IntPoints = metricValue
+				metrics[ind].FloatPoints = nil
+			}
+		} else {
+			return nextCollectionTime, nil, errors.New("No match found for metric regexp: " + metricConfig.Regex)
+		}
```

Comment:
We should probably keep a slice of errors that we then compile together. We try to get all the metrics we can and only return errors at the end. We can use:

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/errors/errors.go#L31